### PR TITLE
Refactorings, fixes and cleanups

### PR DESCRIPTION
**Pull request by pawel-krupinski-guardian**
Updating github actions to simplify caching.
Simplifying log lines format.
Bringing CatalogService in from membership-common.
Fixing compiler warnings.
Removing usages of SafeLogger in favour of StrictLogging. Introducing SanitizerLogger. (SafeLogger made it confusing to figure out where the logging happened)

### CODE REVIEW

**Comment by pawel-krupinski-guardian** [comment]
[Fixes and cleanups in members-data-api](https://trello.com/c/Iju9Umew/4000-fixes-and-cleanups-in-members-data-api)

File: cloudformation/membership-attribute-service.yaml [hunk 1]
```diff
@@ -236,7 +236,7 @@ Resources:
               aws s3 cp s3://gu-membership-attribute-service-dist/membership/${Stage}/membership-attribute-service/membership-attribute-service_1.0-SNAPSHOT_all.deb /tmp
               dpkg -i /tmp/membership-attribute-service_1.0-SNAPSHOT_all.deb
               mkdir -p /etc/gu
-              aws --region ${AWS::Region} s3 cp s3://gu-reader-revenue-private/membership/members-data-api/${Stage}/members-data-api.private.conf /etc/gu
```
Comment (rupertbates) [inline cloudformation/membership-attribute-service.yaml:239]: Is this not needed?

**Review by rupertbates — APPROVED** [review]
Just one question, otherwise 👍

File: cloudformation/membership-attribute-service.yaml (hunk 1 above)
Comment (pawel-krupinski-guardian) [reply cloudformation/membership-attribute-service.yaml:239]: It's not - it was already not on the s3 cp three lines up.

**Pull request merged**